### PR TITLE
Fixup GTK 4 dependents crashing in tests

### PR DESCRIPTION
**Pull request by jtojnar**
###### Description of changes

Followup to https://github.com/NixOS/nixpkgs/pull/207322.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 207546` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>gnome-builder</li>
    <li>vala-language-server</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>ashpd-demo</li>
    <li>bada-bib</li>
    <li>bottles</li>
    <li>bottles-unwrapped</li>
    <li>citations</li>
    <li>fractal-next</li>
    <li>giara</li>
    <li>gnome-text-editor</li>
    <li>gnome.gnome-calculator</li>
    <li>gnome.gnome-maps</li>
    <li>gtksourceview5</li>
    <li>icon-library</li>
    <li>libshumate</li>
    <li>lxi-tools-gui</li>
    <li>pantheon.elementary-iconbrowser</li>
    <li>paper-note</li>
    <li>pods</li>
    <li>shortwave</li>
    <li>textpieces</li>
    <li>zrythm</li>
  </ul>
</details>

**Comment by trofi** [comment]
`gnome-builder` fails with similar  `20/20 test-shortcuts           FAIL            0.13s   killed by signal 5 SIGTRAP`. needs same change?

**Review by trofi — APPROVED** [review]

File: pkgs/development/libraries/gtksourceview/5.x.nix [hunk 1]
```diff
@@ -86,10 +86,12 @@ stdenv.mkDerivation rec {
   checkPhase = ''
     runHook preCheck
 
-    XDG_DATA_DIRS="$XDG_DATA_DIRS:${shared-mime-info}/share" \
-    xvfb-run -s '-screen 0 800x600x24' dbus-run-session \
-      --config-file=${dbus}/share/dbus-1/session.conf \
-      meson test --no-rebuild --print-errorlogs
+    env \
```
Comment (ncfavier) [inline pkgs/development/libraries/gtksourceview/5.x.nix:89]: (Why) do we need to use `env`?

**Review by ncfavier — APPROVED** [review]

File: pkgs/development/libraries/gtksourceview/5.x.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/development/libraries/gtksourceview/5.x.nix:89]: It looks better when all environment variables are aligned the same.

Comment (ncfavier) [reply pkgs/development/libraries/gtksourceview/5.x.nix:89]: You could have the exact same alignment without this line! Though maybe it looks a bit weird (?)

**Review by dotlambda — APPROVED** [review]
`nixosTests.gnome` works again

**Pull request merged**